typescript/sdk/src/token/Token.ts

```diff
@@ -172,10 +176,10 @@ export class Token implements IToken {
    * @throws If multiProvider does not contain this token's chain.
    * @throws If token is an NFT (TODO NFT Adapter support)
    */
-  getHypAdapter(
+  async getHypAdapter(
     multiProvider: MultiProtocolProvider<{ mailbox?: Address }>,
     destination?: ChainName,
-  ): IHypTokenAdapter<unknown> {
+  ): Promise<IHypTokenAdapter<unknown>> {
```

Comment:
this feels like a pretty major change so would like for someone closer than me to the SDK/CLI/registry to have an opinion here. E.g. I imagine this will require lots of changes on the warp UI, some of which may not be easily possible if this function is called by non-async functions

Other ideas that come to mind:
1. Leave the constructor as non-async, and have a state variable internal to the SealevelHypCollateralAdapter that is initially uninitialized. Then we make whatever functions we have that require the knowledge of the SPL token program ID to be async, and it has logic to have a one-time initialization of the SPL token type or to use the already initialized value
2. Add the SPL token type to the config schema, so we don't need to interact with the chain at all

Reply:
I think the `WarpCore` api mostly encapsulates this for other consumers like the warp UI but curious to get @Xaroz 's thoughts

Reply:
agree though that there are other less breaking approaches

Reply:
it sounds like @xeno097 tested this with a local UI build

Reply:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/sdk/src/token/adapters/ITokenAdapter.ts#L32-L57
the adapter API is already async everywhere so deriving internally seems like the move

Reply:
Oky I'm going to change the approach to derive internally and retest with a local ui